docs/src/pages/guides/FHIRTerminologyGuide.md

```diff
@@ -0,0 +1,102 @@
+    TranslationOutcome translate(ConceptMap conceptMap, CodeableConcept codeableConcept, TranslationParameters parameters);
+}
+
+```
+
+The `expand `, `lookup`, `validateCode` (CodeSystem), `validateCode` (ValueSet), and `translate` methods support the passing of optional parameters (e.g. `ExpansionParameters`, `LookupParameters`, etc.). Many of the methods also return an "outcome" object. These "parameter" and "outcome" objects are modeled after the input/output parameters specified in the terminology operations from the FHIR Terminology module: [http://hl7.org/fhir/terminology-module.html](http://hl7.org/fhir/terminology-module.html).
+
+The "parameter" objects can be created from a `Parameters` resource:
+
+```java
+Parameters parameters = ...;
+ExpansionParameters expansionParameters = ExpansionParameters.from(parameters);
+```
+
+The "outcome" objects can be converted to a Parameters resource:
+
+```java
+LookupOutcome outcome = ...;
+Parameters parameters = outcome.toParameters();
+```
+
+This bridge to/from the `Parameters` resource enables implementers build both native implementations of the SPI and implementations that access an existing external terminology service.
+
+## Default Terminology Service Provider Implementation
+
+The default implementation of `FHIRTermServiceProvider` ([DefaultTermServiceProvider](https://github.com/IBM/FHIR/master/fhir-term/src/main/java/com/ibm/fhir/term/service/provider/DefaultTermServiceProvider.java)) leverages terminology resources (`CodeSystem`, `ValueSet`, and `ConceptMap`) that are available from the FHIR registry. The only `CodeSystem` resources that are supported are those with complete content (`CodeSystem.content = 'complete'`). `ValueSet` resources are supported if they reference `CodeSystem` resources that have complete content. The default implementation does not support any optional parameters (e.g. `ExpansionParameters`, `TranslationParameters`, `ValidationParameters`, etc.).
+
+## FHIR Terminology Service Singleton facade
+
+The FHIR Terminology Service Singleton facade ([FHIRTermService](https://github.com/IBM/FHIR/master/fhir-term/src/main/java/com/ibm/fhir/term/service/FHIRTermService.java)) loads a `FHIRTermServiceProvider` from the ServiceLoader, if one exists. Otherwise, it will instantiate a `DefaultTermServiceProvider`. Other FHIR server components and user code (Java) that requires terminology capabilities should access them via the `FHIRTermService` singleton facade. Here is an example:
+
+```java
+ValueSet valueSet = ValueSetSupport.getValueSet("http://ibm.com/fhir/ValueSet/vs1");
+Coding coding = Coding.builder()
+        .system(Uri.of("http://ibm.com/fhir/CodeSystem/cs1"))
+        .version(string("1.0.0"))
+        .code(Code.of("a")
+        .display(string("concept a")
+        .build();
+ValidationOutcome outcome = FHIRTermService.getInstance().validateCode(valueSet, coding);
+```
```

Comment:
Is it possible to run multiple terminology service implementations at the same time? and allow to choose which service implementation to call.  just curious. :)

Reply:
Currently no, but in the future, I anticipate that there could be multiple providers and they will advertise their capabilities and FHIRTermService will select the appropriate one based on the request.